### PR TITLE
Add selected layers state to URL hash

### PR DESCRIPTION
**Pull request by willtonkin**
It would be nice to be able to share links and persist map state in the URL.

Currently the position and zoom level are persisted but not layers, selection, etc.

As a first pass and first contribution, I thought it might be nice to add the currently selected layers to the URL hash.

Placement,Placement_map layers are always added back on as default. I didn't want to change this behaviour without discussion

Using a schema of the style:
```
http://localhost:3000/#16.0000/57.6243/14.9257/layers:Placement,Placement_map,Aftermath22,Aftermath23
```

Am following [leaflet-hash-plus's](https://github.com/gregrobson/leaflet-hash-plus#overview) suggested convention for hash metadata and partially followed their [reference implementation](https://github.com/gregrobson/leaflet-hash-plus/blob/main/docs/demo.html#L70)

### CODE REVIEW

**Comment by TimGremalm** [comment]
I just remebered this pull request! Sorry @willtonkin for forgetting to comment or test it.
I seems like a really good improvement that I've been missing. I will try it out!

File: src/loaders/loadGeoJsonFeatureCollections.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const loadGeoJsonFeatureCollections = async (map, styleFunction, groupByP
     const geojsonLayer = L.geoJSON(geojsonData, {
       filter: filterByProperty(groupByProperty, value),
       style: styleFunction(value),
-    }).addTo(map);
```
Comment (lyret) [inline src/loaders/loadGeoJsonFeatureCollections.js:20]: Reasonably if non-breaking. Seems unrelated to the hashing.

**Review by lyret — APPROVED** [review]
Great contribution! :) Totally a functionality we been missing. The code changes is a bit beyond my level of leaflet-understanding to be able to comment on, but the URL is way within reasonably length still (<2k characters). In talks we been discussing generalising the map for different events other than the borderland, and this will work out of the box for that.

File: src/loaders/loadGeoJsonFeatureCollections.js (hunk 1 above)
Comment (willtonkin) [reply src/loaders/loadGeoJsonFeatureCollections.js:20]: Thanks @lyret!

I wanted to centralize the definition of what layers are visible on page load to one place, here in `createMap()`:
https://github.com/willtonkin/borderland-map/blob/add-layers-to-url-hash/src/map.js#L43

`loadGeoJsonFeatureCollections()` is only called by `createMap()`, which fires on page load to load  `Placement` and `Placement_map`.

`.addTo(map)` adds the layers to the map, making them visible.

My hope would be to move us towards only using this line to add layers on page load:
https://github.com/willtonkin/borderland-map/blob/master/src/map.js#L178

For this to be the case, we would need to remove all other calls to `.addTo(map)`

However, since the behavior is to always load `Placement` and `Placement_map` regardless of whether they're in the hash `layer` array or not, this PR doesn't change that.

I'm happy to walk this back if needed, as the behavior should be the same with or without the line.

I'll await your feedback @lyret before merging 😊

Here is an additional related change where another `.addTo(map)`  was removed for the same reason:
https://github.com/theborderland/map/pull/8/files#diff-49388a0ca466f7c0f2b6502d32934815a6fad08beb14688f86e75aabd035ca9dL82

Comment (willtonkin) [reply src/loaders/loadGeoJsonFeatureCollections.js:20]: Ah, I just spotted you'd approved the change so I will merge. Thanks! 🙇🏻

Comment (lyret) [reply src/loaders/loadGeoJsonFeatureCollections.js:20]: Thank you @willtonkin :) fantastic explanation. This seems very reasonable to me, a lot of the front end code was done was rushed to get it out done before the borderland and any help refactoring is much appreciated. We are loosely talking about how to move the project forward for next year - would love to have you join these talks if you’re interested in joining?

Ping @TimGremalm @Gusbod

Comment (willtonkin) [reply src/loaders/loadGeoJsonFeatureCollections.js:20]: I would love to collab on this! 🤓  I've dropped you a DM in Discord to connect

**Comment by TimGremalm** [comment]
I've tested it and it works great. Thanks for your contribution @willtonkin
You're welcome to join the development for next year if you would like :)
